File: tests/Homework1.Tests/CalculatorTests.cs
using System;
using Xunit;

namespace Homework1.Tests
{
    public class CalculatorTests
    {
        [Theory]
        [InlineData(1, "+", 2, 3)]
        [InlineData(2, "*", 2, 4)]
        [InlineData(2, "-", 2, 0)]
        [InlineData(2, "/", 2, 1)]
        [InlineData(2, ".", 2, 0)]
        public void Calculate_ValidOperations(int val1, string operation, int val2, int expected)
        {
            var actual = Homework2.Calculator.Calculate(operation, val1, val2);
            Assert.Equal(expected, actual);
        }

        [Theory]
        [InlineData(0, "/", 0, Int32.MaxValue)]
        public void Calculate_DivideByZero(int val1, string operation, int val2, int expected)
        {
            var actual = Homework2.Calculator.Calculate(operation, val1, val2);
            Assert.Equal(expected, actual);
        }
    }
}

File: tests/Homework4.Tests/CalculatorTests.fs
module CalculatorTests

open System
open Homework4
open Homework4.Calculator
open Xunit

[<Theory>]
[<InlineData(1, "+", 2, Calculator.result.Ok, 3)>]
[<InlineData(1, "-", 2, Calculator.result.Ok, -1)>]
[<InlineData(1, "*", 2, Calculator.result.Ok, 2)>]
[<InlineData(2, "/", 2, Calculator.result.Ok, 1)>]
let ``Calculate valid arguments`` (val1: int, operation: string, val2: int, exceptedResult :result, expectedValue: int) =
    let excepted = (exceptedResult, expectedValue)
    let actual = Calculate val1 operation val2
    Assert.Equal(excepted, actual)
    
    
[<Theory>]
[<InlineData(1, ".", 2, Calculator.result.UndefinedOperation, 0)>]
[<InlineData(1, "/", 0, Calculator.result.DivideByZero, 0)>]
let ``Calculate invalid arguments`` (val1: int, operation: string, val2: int, exceptedResult :result, expectedValue: int) =
    let excepted = (exceptedResult, expectedValue)
    let actual = Calculate val1 operation val2
    Assert.Equal(excepted, actual)